pkg/util/oas/oas.go

```diff
@@ -16,7 +16,7 @@ func ParseOAS2(spec []byte) (*openapi2.T, error) {
 	swaggerObj := &openapi2.T{}
 	err := json.Unmarshal(spec, swaggerObj)
 	if err != nil {
-		return nil, err
+		return nil, errors.New("unable to parse OAS2 specification")
```

Comment:
I think we should just log an error here, with the details, and return as is. Same with the other one.